src/MangaJet/app/src/main/java/com/mangajet/mangajet/game/GameActivity.kt

```diff
@@ -0,0 +1,183 @@
+        val width: Int = bitmap.width / xCount
+        val height: Int = bitmap.height / yCount
+        for (y in 0 until xCount)
+            for (x in 0 until yCount)
+                bitmaps.add(Bitmap.createBitmap(bitmap, x * width, y * height, width, height))
+
+        return bitmaps
+    }
+
+    // Function to shuffle board
+    private fun shuffle(){
+        var n = tilesNum
+        while(n > 1){
+            var randomNum = Random.nextInt(n--)
+            var tmp = tiles[randomNum]
+            tiles[randomNum] = tiles[n]
+            tiles[n] = tmp
+        }
+        if(!isSolvable())
+            shuffle()
+    }
+
+    // Function to set board after shuffle
+    private fun setBoard(){
+        emptyX = 2
+        emptyY = 2
+        for(i in 0 until group.childCount - 1){
+            buttons[i / (emptyX + 1)][i % ((emptyX + 1))]?.setImageBitmap(bitmaps[tiles[i] - 1])
+            buttons[i / (emptyX + 1)][i % ((emptyX + 1))]?.scaleType = ImageView.ScaleType.FIT_XY
+
+            posMatrix[i / (emptyX + 1)][i % ((emptyX + 1))] = ((tiles[i] - 1) / (emptyX + 1)) *
+                    "10".toInt() +  ((tiles[i] - 1) % (emptyX + 1))
+        }
+
+        buttons[emptyX][emptyY]?.alpha = 0F
+
+    }
+
+    // Function to check if can solve this puzzle
+    private fun isSolvable(): Boolean {
+        var inversions = 0
+        for(i in 0 until tilesNum){
+            var j = 0
+            while(j < i)
+                if(tiles[j++] > tiles[i])
+                    inversions++
+        }
+        return inversions%2 == 0
+    }
+
+    // Function to check if puzzle is solved
+    private fun checkWin(): Boolean {
+        if(emptyY == 2 && emptyX == 2) {
+            var tmp = 0
+            for(i in posMatrix.indices)
+                for (j in 0 until posMatrix[i].size)
+                    if (posMatrix[i][j] < tmp)
+                        return false
+                    else
+                        tmp = posMatrix[i][j]
+        }
+        else
+            return false
+        return true
+
+    }
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Logger.log("Game activity opened")
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_game)
+
+        // Init
+        group = findViewById<ViewGroup>(R.id.group)
+        buttons = Array(emptyX + 1) {
+            arrayOfNulls<ImageButton>(
+                emptyY + 1
+            )
+        }
+        posMatrix = Array(emptyX + 1) { IntArray(emptyX + 1)}
+            for(i in 0 until group.childCount){
+            val child = group.getChildAt(i)
+            if (child is ImageButton)
+                buttons[i / (emptyX + 1)][i % ((emptyX + 1))] = child
+        }
+        tiles = IntArray(tilesNum + 1)
+        for(i in 0 until group.childCount - 1)
+            tiles[i] = i + 1
+
+        // Picture to 9 parts and to buttons
+        val bitmap = resources.getDrawable(R.drawable.anime_girl).toBitmap()
```

Comment:
Maybe we should use some pics from https://konachan.net or https://rule34.paheal.net